src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5406,13 +5487,89 @@ private void BindMemberAccessReportError(BoundMethodGroup node, DiagnosticBag di
+                        new CSharpRequiredLanguageVersion(LanguageVersion.CSharp7_1));
+                }
                 else
                 {
                     Error(diagnostics, ErrorCode.ERR_NoSuchMemberOrExtension, name, boundLeft.Type, plainName);
                 }
             }
         }
 
+        /// <summary>
+        /// When reporting a missing member error on a tuple type, determine whether that member
+        /// could be an element with an inferred name (had the language version been above 7.1).
+        /// </summary>
+        private bool IsMissingMemberAnInferredTupleName(BoundExpression boundLeft, string plainName)
+        {
+            Debug.Assert(boundLeft.Type.IsTupleType);
+            if (Compilation.LanguageVersion >= LanguageVersion.CSharp7_1)
+            {
+                return false;
+            }
+
+            var declarations = boundLeft.Type.DeclaringSyntaxReferences;
+            if (declarations.Count() != 1)
+            {
+                return false;
+            }
+
+            var syntax = declarations[0].GetSyntax();
+            if (syntax.Kind() != SyntaxKind.TupleExpression)
+            {
+                return false;
+            }
+
+            var arguments = ((TupleExpressionSyntax)syntax).Arguments;
+            var (elementNamesArray, inferredArray) = ExtractTupleElementNames(arguments);
+
+            for (int i = 0; i < arguments.Count; i++)
+            {
+                if (elementNamesArray[i]?.Equals(plainName, StringComparison.Ordinal) == true && inferredArray[i])
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private (ImmutableArray<string> elementNamesArray, ImmutableArray<bool> inferredArray) ExtractTupleElementNames(
+            SeparatedSyntaxList<ArgumentSyntax> arguments)
+        {
+            int numElements = arguments.Count;
+            var uniqueFieldNames = PooledHashSet<string>.GetInstance();
+            ArrayBuilder<string> elementNames = null;
+            ArrayBuilder<string> inferredElementNames = null;
+
+            for (int i = 0; i < numElements; i++)
+            {
+                ArgumentSyntax argumentSyntax = arguments[i];
+                IdentifierNameSyntax nameSyntax = argumentSyntax.NameColon?.Name;
+
+                string name = null;
+                string inferredName = null;
+
+                if (nameSyntax != null)
+                {
+                    name = nameSyntax.Identifier.ValueText;
+                }
+                else
+                {
+                    inferredName = InferTupleElementName(argumentSyntax.Expression);
+                }
+                CollectTupleFieldMemberName(name, i + 1, numElements, ref elementNames);
+                CollectTupleFieldMemberName(inferredName, i + 1, numElements, ref inferredElementNames);
```

Comment:
Each use of `CollectTupleFieldMemberName()` passes `i + 1`. Should the method take an offset rather than position?